DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -74,13 +73,13 @@ public IActionResult EditDetails()
                 SelectListHelper.MapOptionsToSelectListItemsWithSelectedText(jobGroups, delegateUser?.JobGroupName);
             ViewBag.CustomFields = GetCustomFieldsWithDelegateAnswers(delegateUser);
 
-            var model = new EditDetailsViewModel(adminUser, delegateUser, jobGroups);
+            var model = new EditDetailsViewModel(adminUser, delegateUser, jobGroups, application);
 
             return View(model);
         }
 
         [NoCaching]
-        [HttpPost]
+        [HttpPost("EditDetails")]
         public IActionResult EditDetails(EditDetailsViewModel model, string action)

```

Comment:
Currently we're using `EditDetailsViewModel` as both
* The data to configure the view
* The data structure for holding the form values

Let's decouple those by
* Extracting the form data into another class called `EditDetailsFormData`
* Attaching this the form data to `EditDetailsViewModel` as a property
* Changing the type of `model` to `EditDetailsFormData` and accepting `application` as a separate parameter.

And let's do the same with other forms.
(The quickest way of making the change may be to start by renaming `EditDetailsViewModel` to `EditDetailsFormData`, as that will update all the existing usages - I expect most of them will be using the form data but I'm not sure)